File: src/com/Torvald/Terrarum/ConsoleCommand/Zoom.java
package com.Torvald.Terrarum.ConsoleCommand;

import com.Torvald.Terrarum.Game;

/**
 * Created by minjaesong on 16-01-25.
 */
public class Zoom implements ConsoleCommand {
    @Override
    public void execute(String[] args) {
        if (args.length == 2) {

            float zoom;
            try {
                zoom = new Float(args[1]);
            }
            catch (NumberFormatException e) {
                new Echo().execute("Wrong number input.");
                return;
            }

            if (zoom < Game.ZOOM_MIN) {
                zoom = Game.ZOOM_MIN;
            }
            else if (zoom > Game.ZOOM_MAX) {
                zoom = Game.ZOOM_MAX;
            }

            Game.screenZoom = zoom;

            System.gc();

            new Echo().execute("Set screen zoom to " + String.valueOf(zoom));
        }
        else {
            printUsage();
        }
    }

    @Override
    public void printUsage() {
        new Echo().execute("Usage: zoom [zoom]");
    }
}

File: src/com/torvald/terrarum/console/Zoom.kt
package com.torvald.terrarum.console

import com.torvald.terrarum.Terrarum

/**
 * Created by minjaesong on 16-01-25.
 */
class Zoom : ConsoleCommand {
    override fun execute(args: Array<String>) {
        if (args.size == 2) {

            var zoom: Float
            try {
                zoom = args[1].toFloat()
            }
            catch (e: NumberFormatException) {
                Echo().execute("Wrong number input.")
                return
            }

            if (zoom < Terrarum.game.ZOOM_MIN) {
                zoom = Terrarum.game.ZOOM_MIN
            }
            else if (zoom > Terrarum.game.ZOOM_MAX) {
                zoom = Terrarum.game.ZOOM_MAX
            }

            Terrarum.game.screenZoom = zoom

            System.gc()

            Echo().execute("Set screen zoom to " + zoom.toString())
        }
        else {
            printUsage()
        }
    }

    override fun printUsage() {
        Echo().execute("Usage: zoom [zoom]")
    }
}
